packages/suite/src/components/suite/layouts/SuiteLayout/PageHeader/HeaderActions.tsx

```diff
@@ -0,0 +1,165 @@
+    const account = useSelector(selectSelectedAccount);
+    const routerParams = useSelector(state => state.router.params) as WalletParams;
+    const selectedAccount = useSelector(state => state.wallet.selectedAccount);
+
+    const dispatch = useDispatch();
+    const layoutSize = useSelector(state => state.resize.size);
+    const isMobileLayout = layoutSize === 'TINY';
+
+    const network = getNetwork(routerParams?.symbol || '');
+    const networkType = account?.networkType || network?.networkType || '';
+    const accountType = account?.accountType || routerParams?.accountType || '';
+
+    const goToWithAnalytics = (...[routeName, options]: Parameters<typeof goto>) => {
+        if (account?.symbol) {
+            analytics.report({
+                type: EventType.AccountsActions,
+                payload: { symbol: account.symbol, action: routeName },
+            });
+        }
+        dispatch(goto(routeName, options));
+    };
+
+    const additionalActions: ActionItem[] = [
+        {
+            id: 'wallet-add-token',
+            callback: () => {
+                if (account?.symbol) {
+                    analytics.report({
+                        type: EventType.AccountsActions,
+                        payload: { symbol: account.symbol, action: 'add-token' },
+                    });
+                }
+                dispatch(openModal({ type: 'add-token' }));
+            },
+            title: <Translation id="TR_TOKENS_ADD" />,
+            isHidden: !['ethereum'].includes(networkType),
+        },
+        {
+            id: 'wallet-sign-verify',
+            callback: () => {
+                goToWithAnalytics('wallet-sign-verify', { preserveParams: true });
+            },
+            title: <Translation id="TR_NAV_SIGN_AND_VERIFY" />,
+            icon: 'SIGNATURE',
+            // show dots when acc missing as they are hidden only in case of XRP
+            isHidden: !account ? false : !hasNetworkFeatures(account, 'sign-verify'),
+        },
+    ];
+
+    const visibleAdditionalActions = additionalActions?.filter(action => !action.isHidden);
+
+    const isCoinmarketAvailable = !['coinjoin'].includes(accountType);
+    const isAccountLoading = selectedAccount.status === 'loading';
+
+    const ButtonComponent = isMobileLayout ? IconButton : Button;
+
+    return (
+        <Container>
+            {!!visibleAdditionalActions?.length && (
+                <AppNavigationTooltip>
+                    <Dropdown
+                        alignMenu="bottom-right"
+                        isDisabled={isAccountLoading}
+                        data-test="@wallet/menu/extra-dropdown"
+                        items={[
+                            {
+                                key: 'extra',
+                                options: visibleAdditionalActions.map<DropdownMenuItemProps>(
+                                    item => {
+                                        const { id, title } = item;
```

Comment:
[optional, nit]: not doing this spread and using it as `item.id|title` would allow use short version of funcion and reducing the code size